### PR TITLE
Descriptor controllers

### PR DESCRIPTION
**Pull request by bamandel**
# Pull Request template

**Not ready to merge**

**Changes proposed in this pull request:**

* Allow the UI to be created dynamically through descriptors
* descriptors now offer options to add startup properties and UI
* descriptor rework

### CODE REVIEW

File: src/main/java/com/blackducksoftware/integration/alert/common/enumeration/DescriptorConfigType.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package com.blackducksoftware.integration.alert.common.enumeration;
+
+public enum DescriptorConfigType {
+    GLOBAL_CONFIG,
```
Comment (psantos1113) [inline src/main/java/com/blackducksoftware/integration/alert/common/enumeration/DescriptorConfigType.java]: add CHANNEL_ as a prefix

File: src/main/java/com/blackducksoftware/integration/alert/common/enumeration/DescriptorConfigType.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    DISTRIBUTION_CONFIG,
```
Comment (psantos1113) [inline src/main/java/com/blackducksoftware/integration/alert/common/enumeration/DescriptorConfigType.java]: add CHANNEL_ as a prefix

File: src/main/java/com/blackducksoftware/integration/alert/web/channel/actions/ChannelDistributionConfigActions.java [hunk 3]
```diff
@@ -251,18 +249,27 @@ public String validateConfig(final CommonDistributionConfig restModel, final Cha
 
     @Override
     public String testConfig(final CommonDistributionConfig restModel, final ChannelDescriptor descriptor) throws IntegrationException {
-        return distributionChannelManager.sendTestMessage(restModel, descriptor);
+        // TODO second expression is a current workaround to allow Slack to have a global config to display in UI and still let tests work properly.
+        if (descriptor.getConfig(DescriptorConfigType.GLOBAL_CONFIG) != null && (descriptor.getConfig(DescriptorConfigType.GLOBAL_CONFIG).getRepositoryAccessor() != null)) {
```
Comment (psantos1113) [inline src/main/java/com/blackducksoftware/integration/alert/web/channel/actions/ChannelDistributionConfigActions.java]: We should change the About Controller to return the list of channels and add about to the UI.  Leave this TODO here but that would solve that issue.  It's probably a day or two at the most to add a new about screen.

Comment (bamandel) [reply src/main/java/com/blackducksoftware/integration/alert/web/channel/actions/ChannelDistributionConfigActions.java]: I've added an issue to Jira to track this

**Pull request merged**